opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -296,277 +196,151 @@ class Restart : private boost::noncopyable
-    }
-
-    // no destructor in this class as ERT takes care of freeing the
-    // handle as part of freeing the solution handle!
-
-    ecl_sum_tstep_type *ertHandle() const
-    { return ertHandle_; };
-
-private:
-    ecl_sum_tstep_type *ertHandle_;
-};
-
-
 /**
  * Initialization file which contains static properties (such as
  * porosity and permeability) for the simulation field.
  */
-class Init : private boost::noncopyable
-{
+class Init {
 public:
-    Init(const std::string& outputDir,
-         const std::string& baseName,
-         int writeStepIdx,
-         IOConfigConstPtr ioConfig) : egridFileName_(outputDir,
-                                                     baseName,
-                                                     ECL_EGRID_FILE,
-                                                     writeStepIdx,
-                                                     ioConfig->getFMTOUT())
-    {
-        bool formatted = ioConfig->getFMTOUT();
-
-        FileName initFileName(outputDir,
-                              baseName,
-                              ECL_INIT_FILE,
-                              writeStepIdx,
-                              formatted);
-
-        ertHandle_ = fortio_open_writer(initFileName.ertHandle(),
-                                        formatted,
-                                        ECL_ENDIAN_FLIP);
-    }
+    Init( const std::string& outputDir,
+          const std::string& baseName,
+          int writeStepIdx,
+          const IOConfig& ioConfig ) :
+        egrid( outputDir,
+               baseName,
+               ECL_EGRID_FILE,
+               writeStepIdx,
+               ioConfig.getFMTOUT() ),
+        init( fortio_open_writer(
+                    FileName( outputDir,
+                        baseName,
+                        ECL_INIT_FILE,
+                        writeStepIdx,
+                        ioConfig.getFMTOUT() ).ertHandle(),
+                    ioConfig.getFMTOUT(),
+                    ECL_ENDIAN_FLIP ) )
 
-    ~Init()
-    { fortio_fclose(ertHandle_); }
+    {}
 
-    void writeHeader(int numCells,
-                     const int* compressedToCartesianCellIdx,
-                     const SimulatorTimerInterface& timer,
-                     Opm::EclipseStateConstPtr eclipseState,
-                     const PhaseUsage uses,
-                     const NNC& nnc = NNC())
+    void writeHeader( int numCells,
+                      const int* compressedToCartesianCellIdx,
+                      time_t current_posix_time,
+                      const EclipseState& es,
+                      int ert_phase_mask,
+                      const NNC& nnc = NNC() )
     {
-        auto dataField = eclipseState->get3DProperties().getDoubleGridProperty("PORO").getData();
-        restrictAndReorderToActiveCells(dataField, numCells, compressedToCartesianCellIdx);
+        auto dataField = es.get3DProperties().getDoubleGridProperty("PORO").getData();
```

Comment:
There is certainly room for improvement in this code; the current ambition is "just" to get it to compile and work with the opm-core <-> opm-output depency changed (i.e. opm-output should *not* depend on opm-core). When 
that is fixed we can continue with improvements, but your initial claim:

>note that outputing anything directly from EclipseState's grid properties should be considered to be bad style because 

Is just plain wrong. 

You are right that the simulator uses `PORV` - which is either initialized from cell volumes, `PORO`and `NTG` or entered explicitly in the grid. We should output `PORV` - from the `Eclipse3DProperties` array[1]; and if present in the grid we should output `PORO` and also `NTG`. But the fact the simulator uses `PORV` does not in any way warrant the claim that we should not output `PORO`.

[1]: Actually the `PORV` vector is somewhat special in the INIT file, that vector *should* be written as a vector of `nx * ny * nz` elements - where zero values can be used to infer the active-inactive cells without using a grid. We do not have that special casing in place yet (I think) - but that should be trivial to add.